docs/Contributing.asciidoc

```diff
@@ -638,6 +638,29 @@ standard output. This makes use of `Test::Output` simple but it should be
 taken care that the test output is not cluttered by log messages which can be
 quite irritating.
 
+=== Test runtime limits
+
+The test modules use `OpenQA::Test::TimeLimit` to introduce a test module
+specific timeout. Consider lowering the timeout value whenever a test module is
+optimized in runtime. If the timeout is hit the test module normally aborts
+with a corresponding message. Please be aware of the exception when the timeout
+triggers after the actual test part of a test module has finished but not all
+involved processes have finished. In this case the output can look like
```

Comment:
Maybe it is worth mentioning the `END` block here. Even without having any sub processes we observed the behavior when the timeout is reached while the `END` block is executed.